### PR TITLE
Fix abnormal cpu usage while syncing

### PR DESCRIPTION
**Pull request by sgaragagghu**
Fix abnormal cpu usage while syncing (downloader side) #254

### CODE REVIEW

File: client/block.c [hunk 1]
```diff
@@ -539,6 +536,29 @@ static int add_block_nolock(struct xdag_block *newBlock, xdag_time_t limit)
 		goto end;
 	}
 
+	for(i = 1, j = 0; i < XDAG_BLOCK_FIELDS; ++i, j++) {
+		if(1 << i & (inmask | outmask)) {
+			blockRefs[j] = block_by_hash(newBlock->field[i].hash);
```
Comment (jonano614) [inline client/block.c]: Why do you need variable j here if you can use `i - 1` ?

File: client/sync.c [hunk 2]
```diff
@@ -40,7 +44,7 @@ static int push_block(struct xdag_block *b, void *conn, int nfield, int ttl)
 
 	xdag_hash(b, sizeof(struct xdag_block), hash);
 	
-	pthread_mutex_lock(&g_sync_hash_mutex);
+	//pthread_mutex_lock(&g_sync_hash_mutex);
```
Comment (jonano614) [inline client/sync.c]: Do you still need these commented lines?

File: client/block.c [hunk 3]
```diff
@@ -468,8 +468,7 @@ static int add_block_nolock(struct xdag_block *newBlock, xdag_time_t limit)
 	int signinmask = 0, signoutmask = 0;
 	int inmask = 0, outmask = 0;
 	int verified_keys_mask = 0, err, type;
-	struct block_internal tmpNodeBlock, *blockRef = NULL, *blockRef0 = NULL;
-	struct block_internal* blockRefs[16]= {0};
+	struct block_internal tmpNodeBlock, *blockRef = NULL, *blockRef0 = NULL, *blockRefs[XDAG_BLOCK_FIELDS]= {0};
```
Comment (jonano614) [inline client/block.c]: No, please, do not combine them in one long line.

File: client/block.c [hunk 4]
```diff
@@ -536,14 +535,14 @@ static int add_block_nolock(struct xdag_block *newBlock, xdag_time_t limit)
 		goto end;
 	}
 
-	for(i = 1, j = 0; i < XDAG_BLOCK_FIELDS; ++i, j++) {
+	for(i = 1; i < XDAG_BLOCK_FIELDS; ++i) {
 		if(1 << i & (inmask | outmask)) {
-			blockRefs[j] = block_by_hash(newBlock->field[i].hash);
-			if(!blockRefs[j]) {
+			blockRefs[i] = block_by_hash(newBlock->field[i].hash);
```
Comment (jonano614) [inline client/block.c]: blockRefs[0] is never occupied :)

**Comment by sgaragagghu** [comment]
Fixed

**Pull request merged**